code/modules/cargo/markets/market_item.dm

```diff
@@ -137,3 +148,13 @@
 		return
 	// Uh oh, uplink or item is gone. We will just keep the money and you will not get your order.
 	qdel(src)
+
+/**
+ * Proc that applies secondary effects to objects that are spawned via a market.
+ *
+ * @param spawned_item - Reference to the atom being spawned.
+ * @param legal_status - Is this item considered legal? If illegal, will apply the contraband trait to the spawned item.
+ */
+/datum/market_purchase/proc/post_purchase_effects(atom/spawned_item)
+	if(legallity == MARKET_PRODUCTS_ILLEGAL)
+		ADD_TRAIT(spawned_item, TRAIT_CONTRABAND, INNATE_TRAIT)
```

Comment:
That would make fenced goods and ransomed mobs also contraband, no?

"I'm sorry, Urist, but you've obviously been kidnapped and then ransomed by a buch of pirates, therefore I must sell you back to CentCom."

Reply:
I'd say a good compromise is to treat fenced items as contraband afterwards, but we'll ignore non-objects so that we can't have people getting obliterated for being kidnapped.